### PR TITLE
Replace wget with curl

### PR DESCRIPTION
**Pull request by mitruly**
`wget` is currently a VinylDNS installation dependency that is not available by default on Windows (and possibly other operating systems as well).

Note to reviewers: Tested both `docker-up-api-server.sh` and `docker-up-vinyldns.sh` locally.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ cache:
     - $HOME/.protobuf-2.6.1/src
 
 install:
-  - wget https://github.com/google/protobuf/releases/download/v2.6.1/protobuf-2.6.1.tar.gz
+  - curl -L https://github.com/google/protobuf/releases/download/v2.6.1/protobuf-2.6.1.tar.gz > protobuf-2.6.1.tar.gz
```
Comment (mitruly) [inline .travis.yml]: `-L` points to the location of the file and also follows redirect, whereas `-O` does not appear to follow redirects.

File: bin/docker-up-api-server.sh [hunk 2]
```diff
@@ -8,31 +8,31 @@
 ######################################################################
 
 DIR=$( cd $(dirname $0) ; pwd -P )
-WORK_DIR=$DIR/../target/scala-2.12
-mkdir -p $WORK_DIR
+WORK_DIR="$DIR"/../target/scala-2.12
```
Comment (mitruly) [inline bin/docker-up-api-server.sh]: Also updated variable references to be enclosed in quotes for safety (and to address IntelliJ compiler warnings).

File: bin/docker-up-api-server.sh [hunk 3]
```diff
@@ -8,31 +8,31 @@
...
-echo "Waiting for API to be ready at ${VINYL_URL} ..."
+VINYLDNS_URL="http://localhost:9000"
+echo "Waiting for API to be ready at ${VINYLDNS_URL} ..."
 DATA=""
 RETRY=40
-while [ $RETRY -gt 0 ]
+while [ "$RETRY" -gt 0 ]
 do
-    DATA=$(wget -O - -q -t 1 "${VINYL_URL}/ping")
+    DATA=$(curl -I -s "${VINYLDNS_URL}/ping" -o /dev/null -w "%{http_code}")
```
Comment (mitruly) [inline bin/docker-up-api-server.sh]: - `-o` directs output
- `-w "%{http_code}` extracts the HTTP code
- `-I` head option to not download the body
- `-s` silent mode

File: .travis.yml (hunk 1 above)
Comment (nimaeskandary) [reply .travis.yml]: if pauls pr goes in first don't need this

File: bin/docker-up-vinyldns.sh [hunk 4]
```diff
@@ -27,21 +27,21 @@ do
...
-echo "Waiting for PORTAL to be ready at ${VINYL_URL} ..."
+VINYLDNS_URL="http://localhost:9001"
+echo "Waiting for PORTAL to be ready at ${VINYLDNS_URL} ..."
 DATA=""
 RETRY=40
-while [ $RETRY -gt 0 ]
+while [ "$RETRY" -gt 0 ]
 do
-    DATA=$(wget -O - -q -t 1 "${VINYL_URL}")
+    DATA=$(curl -I -s "${VINYLDNS_URL}/ping" -o /dev/null -w "%{http_code}")
```
Comment (nimaeskandary) [inline bin/docker-up-vinyldns.sh]: The portal doesn't have a ping endpoint

File: bin/docker-up-api-server.sh [hunk 5]
```diff
@@ -8,44 +8,44 @@
...
-echo "Copy all docker to the target directory so we can start up properly and the docker context is small..."
-cp -af $DIR/../docker $WORK_DIR/
+echo "Copy all Docker to the target directory so we can start up properly and the Docker context is small..."
+cp -af "$DIR"/../docker "$WORK_DIR"/
 
-echo "Copy the vinyldns.jar to the api docker folder so it is in context..."
-if [[ ! -f $DIR/../modules/api/target/scala-2.12/vinyldns.jar ]]; then
+echo "Copy the vinyldns.jar to the API Docker folder so it is in context..."
+if [[ ! -f "$DIR"/../modules/api/target/scala-2.12/vinyldns.jar ]]; then
     echo "vinyldns jar not found, building..."
```
Comment (britneywright) [inline bin/docker-up-api-server.sh]: should this be "vinyldns.jar not found, building..."

File: docker/functest/run.sh [hunk 6]
```diff
@@ -16,7 +16,7 @@ do
         let RETRY-=1
         sleep 1
 
-        if [ $RETRY -eq 0 ]
+        if [ "$RETRY" -eq 0 ]
         then
           echo "Exceeded retries waiting for VINYLDNS to be ready, failing"
```
Comment (britneywright) [inline docker/functest/run.sh]: VinylDNS

**Pull request merged**